src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -230,4 +312,96 @@ private PostContingencyResult runPostContingencySimulation(LfNetwork network, Ac
                                          postContingencyNetworkResult.getBusResults(),
                                          postContingencyNetworkResult.getThreeWindingsTransformerResults());
     }
+
+    private Optional<OperatorStrategyResult> runActionSimulation(LfNetwork network, AcLoadFlowContext context, Contingency contingency, OperatorStrategy operatorStrategy,
+                                                                 LimitViolationManager preContingencyLimitViolationManager,
+                                                                 SecurityAnalysisParameters.IncreasedViolationsParameters violationsParameters,
+                                                                 LimitViolationsResult postContingencyLimitViolations, Map<String, LfAction> lfActionById,
+                                                                 PreContingencyNetworkResult preContingencyNetworkResult, boolean createResultExtension,
+                                                                 List<String> allSwitchesToCloseIds) {
+        OperatorStrategyResult operatorStrategyResult = null;
+
+        if (checkCondition(operatorStrategy, postContingencyLimitViolations)) {
+            LOGGER.info("Start operator strategy {} after contingency '{}' simulation on network {}", operatorStrategy.getId(),
+                    operatorStrategy.getContingencyId(), network);
+
+            List<LfAction> operatorStrategyLfActions = operatorStrategy.getActionIds().stream()
+                    .map(id -> {
+                        LfAction lfAction = lfActionById.get(id);
+                        if (lfAction == null) {
+                            throw new PowsyblException("Action '" + id + "' of operator strategy '" + operatorStrategy.getId() + "' not found");
+                        }
+                        return lfAction;
+                    })
+                    .collect(Collectors.toList());
+            // FIXME to remove when connectivity will be fixed
+            operatorStrategyLfActions.stream().limit((long) operatorStrategyLfActions.size() - 1).forEach(LfAction::apply);
```

Comment:
As far as I understand this won't be necessary anymore when addEdge and removeEdge will be callable multiple tile with same branch without any exception?

Reply:
It is not so simple: it is also for performance issues. As we can have a list of actions, the connectivity is computed after applying the last one in order to find propagated effects. The FIXME is for 'allSwitchesToCloseIds' argument of the apply, that will be removed once remove/add of edge can be done several times.

Reply:
Connectivity is lazily computed, so we cannot have perf issue

Reply:
Now, refactored to avoid this